File: w1_p5/26.c
#include <stdio.h>
#include <math.h>
int main() {
	int n,i;
	float st[1001],sum,aver,q,w;
	
	while(scanf("%d",&n)==1)
	{
		sum=0;
		q=0;
		w=0;
		
		if(n==0)
		{
			break;
		}
		
		for(i=0; i<n; i++)
		{
			scanf("%f",&st[i]);
			sum = sum + st[i];
		}
		
		aver = round(sum/n*100)/100;
		
		for(i=0; i<n; i++)
		{
			if(st[i] -aver >= 0)
			{
				q= q+st[i]-aver;
			}
			else
			{
				w= w+aver-st[i];
			}
		}
		if(q>w)
		{
		printf("$%.2f\n",w);
		}
		else
			printf("$%.2f\n",q);
		
	}
	
	return 0;
}

File: w1_p5/26.java
import java.io.*;
import java.util.Scanner;
class Main {
	public static void main(String[] args) throws Exception {
		
		Scanner input = new Scanner(System.in);
		
		while(true){
			
			double money=0,give=0,aver;
			int student = input.nextInt();
			double arr[]=new double[student];
			if(student==0){
				break;
			}
			for(int i=0;i<student;i++){
				arr[i]=input.nextDouble();
				money=money+arr[i];
			}
			aver=money/student;
			for(int i=0;i<student;i++){
				if(arr[i]>aver){
					double a = Math.floor((arr[i]-aver)*100)/100;
					give+=a;
				}
			
			}
			System.out.printf("$%.2f\n",give);
			
		}

	}
}